test/lib/roby/test_custom_require.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "roby/test/self"
+require "lib/roby/custom_require.rb"
+
+module Syskit
+    describe KernelExtension do
+        attr_reader :base_dir
+
+        before do
+            @base_dir = make_tmpdir
+        end
+
+        def create_file(*path)
+            FileUtils.mkdir_p File.join(base_dir, *path[0..-2])
+            FileUtils.touch   File.join(base_dir, *path)
+        end
+
+        it "resolves relative paths from registered roby applications" do
+            create_file("models", "compositions", "file.rb")
+            app.register_app(base_dir)
```

Comment:
I would not assume that `make_tmpdir` outputs a "good" app name. Make `@base_dir = File.join(make_tmpdir, "app")`, which will avoid peppering your test with `File.basename(base_dir)`